components/workflow-mgt/org.wso2.carbon.identity.workflow.mgt.ui/src/main/resources/web/workflow-mgt/add-wf-wizard.jsp

```diff
@@ -140,11 +151,22 @@
 
         function submitPage(){
             var workflowForm = document.getElementById("id_workflow");
-            if($('#id_workflow_name').val().length > 0){
+            if(validateWorkflowName()){
                 workflowForm.submit();
-            }else{
+            }
+        }
+
+        function validateWorkflowName(){
+            var workflow_name = $('#id_workflow_name').val();
+            if(workflow_name.length == 0){
                 CARBON.showWarningDialog("<fmt:message key="workflow.error.empty.workflow.name"/>" , null ,null) ;
+                return false;
+            }
+            if(existingWorkflowNames.indexOf(workflow_name + "|") !== -1){
```

Comment:
IMO checking indexOf won't work here. We can say the workflow name is duplicated only if an existing name is exactly equal to the newly adding workflow name.

Reply:
i added separator for the workflow names and here i am checking indexof with that separator... so it is doing exactly checking same name...

Reply:
If we don't have any special reason for this shall we just use "===" without making the logic complex